src/fastcs/backend.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+from collections import defaultdict
+from typing import Callable, cast
+
+from .attributes import AttrCallback, AttrMode, AttrR, AttrW
+from .cs_methods import MethodType
+from .mapping import Mapping, SingleMapping
+
+
+def _get_initial_tasks(mapping: Mapping) -> list[Callable]:
+    initial_tasks: list[Callable] = []
+    initial_tasks.append(mapping.controller.connect)
+    return initial_tasks
+
+
+def _create_periodic_scan_task(period, methods: list[Callable]) -> Callable:
+    async def scan_task() -> None:
+        while True:
+            await asyncio.gather(*[method() for method in methods])
+            await asyncio.sleep(period)
+
+    return scan_task
+
+
+def _get_periodic_scan_tasks(scan_dict: dict[float, list[Callable]]) -> list[Callable]:
+    periodic_scan_tasks: list[Callable] = []
+    for period, methods in scan_dict.items():
+        periodic_scan_tasks.append(_create_periodic_scan_task(period, methods))
+
+    return periodic_scan_tasks
+
+
+def _add_wrapped_scan_tasks(
+    scan_dict: dict[float, list[Callable]], single_mapping: SingleMapping
+):
+    for method_data in single_mapping.methods:
+        if method_data.info.method_type == MethodType.scan:
+            period = method_data.info.kwargs["period"]
+            method = method_data.method
+            scan_dict[period].append(method)
```

Comment:
Because period is only relevant for scan methods.

It's possible to refactor this to use Put / Scan / Command dataclasses and pattern matching, if you're interested in proper typing; I'm just not a fan of optional fields / messy types.

Reply:
Yeah I think that would be a lot cleaner. `Method` base class with `Put`/`Scan`/`Command` child classes instead of a `MethodType` attribute on a generic class instance. And `MethodData` could be pulled into the same class, as I mentioned [here](https://github.com/DiamondLightSource/FastCS/pull/6/files#r1291641894).

Reply:
Done